File: 02-query/src/queryRoles.js
const AWS = require("aws-sdk");
const DynamoDB = new AWS.DynamoDB({ region: "us-east-1" });

DynamoDB.query(
  {
    TableName: "MovieRoles",
    KeyConditionExpression: "Actor = :actor",
    ExpressionAttributeValues: {
      ":actor": { S: "Tom Hanks" },
    },
  },
  function (err, data) {
    if (err) console.log(`Error performing query": ${err}`);
    else
      console.log(
        `Found ${data.Items.length} roles!\n${JSON.stringify(data.Items)}`
      );
  }
);

File: 02-query/python/query_roles.py
import json

import boto3

client = boto3.client("dynamodb", region_name="us-east-1")

try:
    response = client.query(
        TableName="MovieRoles",
        KeyConditionExpression="#actor = :actor",
        ExpressionAttributeNames={"#actor": "Actor"},
        ExpressionAttributeValues={":actor": {"S": "Tom Hanks"}},
    )
    print(
        f"Found {len(response['Items'])} roles!\n{json.dumps(response['Items'], indent=2)}"
    )
except Exception as err:
    print(f"Error fetching roles: {err}")
